src/NuGet.Core/NuGet.Versioning/NuGetVersionFactory.cs

```diff
@@ -11,6 +11,10 @@ namespace NuGet.Versioning
 {
     public partial class NuGetVersion
     {
+        // Cached mappings from string => NuGetVersion. On cache hit, avoids allocations during TryParse.
+        private static Dictionary<string, NuGetVersion> ParsedNuGetVersionsMapping = new Dictionary<string, NuGetVersion>();
```

Comment:
Dictionary doesn't have a `Capacity` property, so I can't check with C# Interactive.   What's the capacity of the dictionary's backing store when you add the first item to the dictionary?

My concern is that the cost of resizing the dictionary will be greater than the cost of overallocating a dictionary. I think we should create the dictionary passing in `MaxEntries` as the initial capacity.

Reply:
First add to a dictionary (on fx at least), allocates a backing store of size 3.

I went ahead and set the initial capacity to ParsedNuGetVersionsMappingMaxEntries, per your suggestion.